src/ASM/ASMs3DLag.C

```diff
@@ -1089,6 +1082,48 @@ bool ASMs3DLag::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 }
 
 
+bool ASMs3DLag::evalSolution (Matrix& sField, const IntegrandBase& integrand,
+                              const RealArray* gpar, bool) const
+{
+  sField.resize(0,0);
+
+  size_t nPoints = gpar[0].size();
+
+  FiniteElement fe(p1*p2*p3);
+  Vector        solPt;
+  Vectors       globSolPt(nPoints);
+  Matrix        dNdu, Xnod, Jac;
+
+  // Evaluate the secondary solution field at each point
+  for (size_t i = 0; i < gpar[0].size(); ++i) {
+    const int iel = this->findElement(gpar[0][i], gpar[1][i], gpar[2][i],
+                                        &fe.xi, &fe.eta, &fe.zeta);
+
+    const IntVec& mnpc = MNPC[iel-1];
```

Comment:
Same, `mnpc` is only used once below. Use `MNPC[iel-1]` directly instead

Reply:
changed